docs-tech/bakery-architecture.md

```diff
@@ -52,3 +52,74 @@ and are pure computations, which traditionally are called "sieves".
+and [here for an interaction](../bakery-integration-tests/src/test/resources/kubernetes/crd/interactions-example.yaml)
+
+Also the specification of the CRDs can be found [here for Baker](../bakery-integration-tests/src/test/resources/kubernetes/crd/crd-baker.yaml)
+and [here for Interaction](../bakery-integration-tests/src/test/resources/kubernetes/crd/crd-interaction.yaml)
+
+### Baker Manifests
+
+The Baker manifest is created by the Bakery Controller from a Baker CRD, and then uses it to mount the recipes that go
+into the Baker Clusters, the naming goes `<name-in-the-crd>-manifest`, and it can be used also to inspect the recipes
+deployed in the namespace.
+
+### Interaction Manifests
+
+The Interaction manifest is created by the Bakery Controller after deploying an interaction from the specification of an
+Interaction CRD, the controller uses the HTTP api of the deployed interaction to request the data structure that specifies
+all the interfaces of the interactions within the Interaction replicas, it aggregates this data structures into the manifest,
+and finally the Baker Clusters can use the Kubernetes API to query for all available interactions in the namespace, effectively
+working as a service discovery mechanism.
+
+## Baker Akka Clusters
+
+When a Baker CRD is created in the namespace, the Bakery Controller creates a Kubernetes Deployment and Service, the deployed
+Pods contain an Akka cluster which keep the state of the baker recipe instances and persists them into a Cassandra cluster, 
+it exposes the Baker API through HTTP, which is accessible from the Kubernetes Service, and it uses the Kubernetes API to discover
+available interactions in the namespace by querying for ConfigMaps with the corresponding interaction manifest labels.
+
+## Interactions
+
+When an Interaction CRD is created in the namespace, the Bakery Controller creates a Kubernetes Deployment and Service,
+from which the deployed Pods use the docker image specified at the Interaction CRD, such image must be built using the
+bakery interaction library, which adds the required http api layer to the interaction implementation from which the
+Bakery Controller can extract the implementations interface for the manifest and from which the Baker Cluster can request 
+the execution of an interaction given some ingredients.
+
+## Configuration mounting
+
+As part of the CRDs of both Bakers and Interactions, one can add the name of a secret and/or a configmap to mount configuration
+within the Bakers and Interaction pods, this is used for example to configure per environment cassandra connectivity or 
+specific configuration required by the interaction implementations. Alternatively the CRD interface allows for environment 
+variable configuration at the CRD level.
+
+## Baker Clients
+
+Baker clients can either consume the HTTP api or use our published bakery-baker-client library to use a baker scala/java 
```

Comment:
perhaps worth adding that the APIs exposed by Baker are 'naked' in the sense that doesn't provide any authn/z 
and this is up to the implementor to arrange